src/ServicePulse.Host/app/js/views/monitored-endpoints/monitored-endpoints.controller.js

```diff
@@ -13,12 +13,9 @@
         });
 
         function updateUI() {
-            monitoringService.getRaw().then(function (data) {
-                $scope.endpoints = data["NServiceBus.Endpoints"];
-
-                timeoutId = $timeout(function () {
-                    updateUI();
-                }, 5000);
+            monitoringService.getEndpoints().subscribe(function (endpoint) {
+                $scope.endpoints = $scope.endpoints || {};
+                $scope.endpoints[endpoint.endpointName] = endpoint;
```

Comment:
Correct. Unless we change the api to return data for a specific endpoint instead of a list of endpoints + data, we'll have to redraw the entire endpoint (including all data) when retrieved.

Unless of course we do some clever hackery around "last updated timestamp" or whatever.